src/models/Util.ts

```diff
@@ -78,6 +80,28 @@ export const Util = {
 
     isTechnicalCivilisation(civilisation: Civilisation): boolean {
         return civilisation.gameVersion === GameVersion.TECHNICAL;
-    }
+    },
+
+    isRandomCiv(civilisation: Civilisation): boolean {
+        return civilisation.name.toUpperCase() === "RANDOM";
+    },
 
+    setRandomCivilisation(playerEvent: PlayerEvent, draftId: string, draftStore: DraftsStore) {
+        const maxCivilisationIndex = Civilisation.ALL.length - 1;
+        const randomCivIndex = Math.floor(Math.random() * maxCivilisationIndex);
+        playerEvent.civilisation = Civilisation.ALL[randomCivIndex];
+
+        const errors = Validator.checkAllValidations(draftId, draftStore, playerEvent);
+
+        if (errors.length === 0) {
+            // random civ is set correctly.
+            playerEvent.civilisation.isRandomlyChosenCiv = true;
+            return;
+        } else {
+            console.log("Errors:", errors, "Randomly chosen civ:", Civilisation.ALL[randomCivIndex]);
+            // recursively try to set random civ
+            playerEvent.civilisation = Civilisation.RANDOM;
```

Comment:
It is a bit unfortunate that we have to reset the event's civilisation to `RANDOM` again here